src/event/source/unix.rs

```diff
@@ -107,20 +90,27 @@ impl EventSource for UnixInternalEventSource {
             for token in self.events.iter().map(|x| x.token()) {
                 match token {
                     TTY_TOKEN => {
-                        let read_count = self.tty_fd.read(&mut self.tty_buffer, TTY_BUFFER_SIZE)?;
-
-                        if read_count > 0 {
-                            self.poll
-                                .poll(&mut additional_input_events, Some(Duration::from_secs(0)))?;
-
-                            let additional_input_available = additional_input_events
-                                .iter()
-                                .any(|event| event.token() == TTY_TOKEN);
-
-                            self.parser.advance(
-                                &self.tty_buffer[..read_count],
-                                additional_input_available,
-                            );
+                        loop {
+                            match self.tty_fd.read(&mut self.tty_buffer, TTY_BUFFER_SIZE) {
+                                Ok(read_count) => {
+                                    if read_count > 0 {
+                                        self.poll
```

Comment:
This line doesn't make a sense because you removed `additional_input_available` variable computation.